students/jinee_han/lesson05/Mail3.py

```diff
@@ -0,0 +1,183 @@
+# --------------------------------
+# 07/14/19 Jinee Han
+# Python Programming Lesson 4
+# Mailroom Part 2
+# ---------------------------------
+
+'''
+update
+Implement comprehensions and exception handling where possible
+'''
+
+import sys, datetime
+
+
+# Donor List
+donor_db = {"Kim Kardasian": [653772.32, 12.17],
+            "Kendal Jenner": [877.33],
+            "Gigi Hadid": [663.23, 43.87, 1.32],
+            "Justin Bieber": [1663.23, 4300.87, 10432.0],
+            "Will Smith": [43.23, 4000.07, 183423.2]
+            }
+
+# Display options
+prompt = "\n".join(("Welcome to the donor list!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a Report",
+          "3 - Send Thank You to All Donors",
+          "4 - Exit",
+          ">>> "))
+
+def send_thank_you_note():
+    '''
+    Check for the user input and send a thank you once user entered
+    :return: nothing
+    '''
+    inputValue = input("Enter a full name. (Type 'list' to see the donor list)")  # Ask the name
+    while inputValue == 'list':
+        display_list()
+        inputValue = input("Enter a full name. (Type 'list' to see the donor list)")
+
+    # Store the donor name with the first character as an upper case letter
+    donor_name = inputValue.title()
+
+    if not donor_present(donor_name):
+        add_donor(donor_name)
+
+    donation_amount = None
+    while donation_amount is None:
+        try:
+            donation_amount = float(input("Please enter the donation amount: "))
+        except ValueError:
+            pass
```

Comment:
This actually works well.  Alternatively, you could give the user a message to enter a valid amount.